controller/RestTestCenter.php

```diff
@@ -189,20 +298,51 @@ public function get()
     }
 
     /**
-     * Get delivery resource from request parameters
-     * @return \core_kernel_classes_Resource
-     * @throws common_exception_MissingParameter
+     * @param array $data
+     * @param bool $isRequired
+     * @return array
+     * @throws common_exception_RestApi
      */
-    private function getLabelFromRequest()
+    protected function getParametersRequestData(array $data, $isRequired = false)
     {
-        return $this->getParameterFromRequest(self::PARAMETER_TEST_CENTER_LABEL);
+        $values = [];
+
+        foreach (array_keys($this->parametersMap) as $name) {
+            if (array_key_exists($name, $data)) {
+                $values[$name] = $data[$name];
+            } else if ($isRequired) {
+                throw new common_exception_RestApi(__('Missed required parameter: `%s`', $name), 400);
+            }
+        }
+        return $values;
+    }
+
+    /**
+     * @param array $values
+     * @return array
+     * @throws common_exception_RestApi|common_exception_Error
+     */
+    protected function prepareRequestData(array $values)
+    {
+        $propertiesValues = [];
+
+        foreach ($values as $name => $value) {
+            if (array_key_exists($name, $this->parametersMap)) {
+                if (empty($value)) {
```

Comment:
Are you sure that empty string can't be a value?

Reply:
I guess it depends on the provided property.
What if you don't need this parameters map?

for the TestCenter extension
```
prepareRequestData(array $values) {
    if (!array_key_exists(self::PARAM_LABEL, $values)) {
         throw new exception ('label required');
    }
    $parameters = [$values[self::PARAM_LABEL]];
    return $parameters;
}
```

for the TaoOffline extension:
```
prepareRequestData(array $values) {
    $parameters = parent::prepareRequestData($values);
    if (!array_key_exists(self::PARAM_ORG_ID, $values)) {
         throw new exception ('label required');
    }
    $parameters[] = [$values[self::PARAM_ORG_ID]];
    return $parameters;
```